### PR TITLE
Update to Elm 0.8.1

### PR DESCRIPTION
**Pull request by rudolf-adamkovic**

### CODE REVIEW

**Comment by andris-zalitis** [comment]
In review

File: AudioBarTests/AudioBarViewTests.swift [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        expect(view.isSeekBackButtonEnabled, false)
+    }
+
+    func testIsSeekBackButtonEnabled2() {
+        model = .readyToPlay(.init(duration: 60, currentTime: 0))
+        expect(view.isSeekBackButtonEnabled, false)
+    }
+
+    func testIsSeekBackButtonEnabled3() {
+        model = .readyToPlay(.init(duration: 60, currentTime: 1))
```
Comment (andris-zalitis) [inline AudioBarTests/AudioBarViewTests.swift:95]: In old tests we had currentTime: 0.01. I suppose there is no difference, but I'd suggest separate commits for test optimisations next time.

File: AudioBarTests/AudioBarViewTests.swift [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        expect(view.isSeekBackButtonEnabled)
+    }
+
+    func testIsSeekBackButtonEnabled4() {
+        model = .readyToPlay(.init(duration: 60, currentTime: 2))
```
Comment (andris-zalitis) [inline AudioBarTests/AudioBarViewTests.swift:100]: Same thing as above but with 0.02.

File: AudioBarTests/AudioBarViewTests.swift [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        expect(view.isSeekBackButtonEnabled)
+    }
+
+    func testIsSeekForwardButtonEnabled1() {
+        model = .readyToPlay(.init(currentTime: nil))
+        expect(view.isSeekForwardButtonEnabled, false)
+    }
+
+    func testIsSeekForwardButtonEnabled2() {
+        model = .readyToPlay(.init(duration: 60, currentTime: 58))
```
Comment (andris-zalitis) [inline AudioBarTests/AudioBarViewTests.swift:110]: In the old tests the values were `duration: 1, currentTime: 0.99`. It'd be easier to review if there were separate commits for updating tests.

File: AudioBarTests/SeekForwardTests.swift [hunk 4]
```diff
@@ -1,77 +0,0 @@
-import XCTest
-@testable import AudioBar
-
-final class SeekForwardTests: XCTestCase {
-
```
Comment (andris-zalitis) [inline AudioBarTests/SeekForwardTests.swift:5]: I was able to find 9 tests (in AudioBarUpdateTests.swift) that seem to correspond to these 11 (values are different for some so it's hard to tell). Make sure you have not accidentally reduced test coverage with this transition.

**Review by andris-zalitis — CHANGES_REQUESTED** [review]
Great, accurate work and beautifully simple tests with the new syntax!

However it was very hard to review. I tried to make sure no tests have been lost in the conversion process and since there is only one commit, bunch of files removed and new test methods have different names, it was... challenging to say the least.

Ideally it would have been one commit per each removed file so that it's traceable that all tests have been moved. Also please refrain from updating test parameters while moving tests, it's better to use separate commits for readability optimisation.

After checking as well as I could, I found one case you should probably take a look at (total number of tests per feature seems to change from 11 to 9).

File: AudioBarTests/AudioBarViewTests.swift (hunk 1 above)
Comment (rudolf-adamkovic) [reply AudioBarTests/AudioBarViewTests.swift:95]: You're right. Will remember that next time. Thanks!

File: AudioBarTests/AudioBarViewTests.swift (hunk 2 above)
Comment (rudolf-adamkovic) [reply AudioBarTests/AudioBarViewTests.swift:100]: You're right. Will remember that next time. Thanks!

File: AudioBarTests/SeekForwardTests.swift (hunk 4 above)
Comment (rudolf-adamkovic) [reply AudioBarTests/SeekForwardTests.swift:5]: Compared to `master`. So, I collapsed the following four tests ...

```swift
func testWhenPlaying1() {
    var model = Model.readyToPlay(.init(isPlaying: true, duration: 60, currentTime: 0))
    let commands = try! Module.update(for: .seekForward, model: &model)
    XCTAssertEqual(model, .readyToPlay(.init(isPlaying: true, duration: 60, currentTime: 15)))
    XCTAssertEqual(commands, [.player(.setCurrentTime(15))])
}

func testWhenPlaying2() {
    var model = Model.readyToPlay(.init(isPlaying: true, duration: 120, currentTime: 60))
    let commands = try! Module.update(for: .seekForward, model: &model)
    XCTAssertEqual(model, .readyToPlay(.init(isPlaying: true, duration: 120, currentTime: 75)))
    XCTAssertEqual(commands, [.player(.setCurrentTime(75))])
}

func testWhenPaused1() {
    var model = Model.readyToPlay(.init(isPlaying: false, duration: 60, currentTime: 0))
    let commands = try! Module.update(for: .seekForward, model: &model)
    XCTAssertEqual(model, .readyToPlay(.init(isPlaying: false, duration: 60, currentTime: 15)))
    XCTAssertEqual(commands, [.player(.setCurrentTime(15))])
}

func testWhenPaused2() {
    var model = Model.readyToPlay(.init(isPlaying: false, duration: 120, currentTime: 60))
    let commands = try! Module.update(for: .seekForward, model: &model)
    XCTAssertEqual(model, .readyToPlay(.init(isPlaying: false, duration: 120, currentTime: 75)))
    XCTAssertEqual(commands, [.player(.setCurrentTime(75))])
}
```

... into the following two tests ...

```swift
func testSeekForward1() {
    model = .readyToPlay(.init(isPlaying: true, duration: 60, currentTime: 0))
    message = .seekForward
    expect(model, .readyToPlay(.init(isPlaying: true, duration: 60, currentTime: 0 + 15)))
    expect(command, .player(.setCurrentTime(0 + 15)))
}

func testSeekForward2() {
    model = .readyToPlay(.init(isPlaying: false, duration: 60, currentTime: 1))
    message = .seekForward
    expect(model, .readyToPlay(.init(isPlaying: false, duration: 60, currentTime: 1 + 15)))
    expect(command, .player(.setCurrentTime(1 + 15)))
}
```

Oops! `isPlaying` doesn't play any role here (just like in `testSeekBack[12]`) and should be removed. Will fix.

**Comment by rudolf-adamkovic** [comment]
@andris-zalitis 

> [...] challenging to say the least.

I will try harder to push smaller, more focused commits. Thanks for your feedback!

Ready again!

**Review by andris-zalitis — APPROVED** [review]

**Pull request merged**